app/assets/javascripts/components/actions/compose.jsx

```diff
@@ -74,8 +74,13 @@ export function mentionCompose(account, router) {
 export function submitCompose() {
   return function (dispatch, getState) {
     dispatch(submitComposeRequest());
+    const status = emojione.shortnameToUnicode(getState().getIn(['compose', 'text'], ''));
+    if (!status || !status.length) {
```

Comment:
It can be simpler. Just do this before the dispatch of the request above, then you don't have to dispatch an extra success